recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -99,19 +99,27 @@ this.PreferenceExperiments = {
   async init() {
     const store = await ensureStorage();
 
-    Object.values(store.data).forEach(experiment => {
-      const {name, branch, expired, preferenceBranchType, preferenceName, preferenceValue} = experiment;
+    Object.values(store.data)
+      .filter(experiment => !experiment.expired)
+      .forEach(experiment => {
+        // Set experiment default preferences, since they don't persist between restarts
+        if (experiment.preferenceBranchType === "default") {
+          DefaultPreferences.set(experiment.preferenceName, experiment.preferenceValue);
+        }
 
-      // Set experiment default preferences, since they don't persist between restarts
-      if (!expired && preferenceBranchType === "default") {
-        DefaultPreferences.set(preferenceName, preferenceValue);
-      }
+        // Check that the current value of the preference is still what we set it to
+        if (Preferences.get(experiment.preferenceName, undefined) !== experiment.preferenceValue) {
+          // if not, stop the experiment, and skip the remaining steps
+          this.stop(experiment.experimentName, false);
```

Comment:
I think the `this` keyword is broken in functions wrapped with `wrapAsync`, but I have a commit lying around that fixes that: https://github.com/Osmose/normandy/commit/2bf562431a6a7e6d2036eefbdb723ab9c2a39894